packages/sanity/src/desk/comments/src/context/selected-path/CommentsSelectedPathProvider.tsx

```diff
@@ -0,0 +1,28 @@
+import React, {useMemo, useState} from 'react'
+import {CommentsSelectedPathContext} from './CommentsSelectedPathContext'
+import {CommentsSelectedPath, CommentsSelectedPathContextValue} from './types'
+
+interface CommentsSelectedPathProviderProps {
+  children: React.ReactNode
+}
+
+export const CommentsSelectedPathProvider = React.memo(function CommentsSelectedPathProvider(
+  props: CommentsSelectedPathProviderProps,
+) {
+  const {children} = props
+  const [selectedPath, setSelectedPath] = useState<CommentsSelectedPath>(null)
```

Comment:
As `setSelectedPath` receives an object literal, any call will always trigger a full re-render of the document pane, regardless of whether any path values (`fieldPath`, `threadId` etc) have changed or not.

This is especially noticeable when clicking between different threads within a field path. On local, this previously used to feel instant but now there's a definite hitch, which is likely due to the entire document form re-rendering.

This issue is also exacerbated by the fact that `setSelectedPath` is being called whenever you click anywhere in the comment. Making a text selection in a comment will trigger two renders, which we probably don't want!

Rather than passing the `useState` set function to the context's value – perhaps we can pass a custom callback which does a shallow comparison before setting internal state?

Reply:
Thanks, I will look into this 👍 

However, in the previous approach, the selected comment was also in the shape of an object, so nothing has really changed in that regard!

> This issue is also exacerbated by the fact that setSelectedPath is being called whenever you click anywhere in the comment. Making a text selection in a comment will trigger two renders, which we probably don't want!

I just pushed a fix to prevent this from happening!

Reply:
The issue here appears to be related to the "on click outside" behavior, which often triggers two state updates when clicking on an item. That is, if one item is already selected and you click on another item, you first update the state due to the click outside the initially selected item, and then a second update occurs because you've clicked on another element.